cli/azd/cmd/middleware/ux.go

```diff
@@ -0,0 +1,32 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+)
+
+// UxMiddleware composes output message of actions that return ActionResults
+type UxMiddleware struct {
+	console input.Console
+}
+
+// Creates a new Ux Middleware instance
+func NewUxMiddleware(console input.Console) Middleware {
+	return &UxMiddleware{
+		console: console,
+	}
+}
+
+// Invokes the middleware and prints out and action result status after the
+// underlying action completes
+func (r *UxMiddleware) Run(ctx context.Context, next NextFn) (*actions.ActionResult, error) {
+	result, err := next(ctx)
+
+	if result != nil {
```

Comment:
FYI - What we have checked in today [considers both the action result and the error](https://github.com/Azure/azure-dev/blob/923799f7d5b14ebce7e5846fb8913c0c4c03f1b6/cli/azd/cmd/root.go#L157-L161).  This was required when I added it because MessageUxItem was responsible for displaying the error.  I'm not sure if this change to stop considering the error again was on purpose or the result of code motion when you rebased, so I'm calling this out here.

Reply:
Will take a look since there was some merge conflicts around this during rebase.

Reply:
Will revert (remove) this UX middleware for now.  I think the intent here was to only have a middleware that ever runs at the root cmd/action.  This concept doesn't exist yet and will add in the current logic until we could support the use case.